qualtran/bloqs/rotations/phase_gradient.py

```diff
@@ -183,16 +197,33 @@ def scaled_val(self, x: int) -> int:
         x_fxp = _fxp(x / 2**x_width, x_width).like(_fxp(0, self.phase_bitsize)).astype(float)
         return int(x_fxp.astype(float) * 2**self.phase_bitsize)
 
-    def apply(self, x: int, phase_grad: int) -> Union[int, Iterable[int]]:
+    def apply(self, *args) -> Union[int, Iterable[int]]:
+        if self.controlled:
+            ctrl, x, phase_grad = args
+            out = self.on_classical_vals(ctrl=ctrl, x=x, phase_grad=phase_grad)
+            return out['ctrl'], out['x'], out['phase_grad']
+
+        x, phase_grad = args
```

Comment:
For all the changes made to `AddIntoPhaseGrad`, you should add tests to `phase_gradient_test.py` to verify the correctness of these changes.